src/evaluate/evaluator.py

```diff
@@ -244,10 +244,12 @@ def compute(
             )
 
         # Prepare pipeline.
-        if (
-            isinstance(model_or_pipeline, PreTrainedModel)
-            or isinstance(model_or_pipeline, TFPreTrainedModel)
-            or isinstance(model_or_pipeline, str)
+        if isinstance(model_or_pipeline, str) or (
+            hasattr(model_or_pipeline, "__class__")
+            and any(
+                cls_name in [parent_cls.__name__ for parent_cls in model_or_pipeline.__class__.__mro__]
+                for cls_name in ["PreTrainedModel", "TFPreTrainedModel"]
+            )
```

Comment:
> I'm not sure if "tensorflow" in sys.modules is the way to do it. Because if the user uses pipeline, that means they don't know in advance what's the model type right?

If `model_or_pipeline` is a pipeline, you don't need to check if it inherits from `TFPreTrainedModel`, so you don't need to import `TFPreTrainedModel` ;)

I'll let @sgugger give his opinion, but I think this is reasonable:

```suggestion
        import transformers
        
        if "tensorflow" in sys.modules:
            # Check if the model if a TF model only if TF has already been imported.
            # Indeed loading `TFPreTrainedModel` may import tensorflow unnecessarily.
            transformers_pretrained_model_classes = (transformers.PreTrainedModel, transformers.TFPreTrainedModel)
        else:
            transformers_pretrained_model_classes = (transformers.PreTrainedModel)

        if (
            isinstance(model_or_pipeline, transformers_pretrained_model_classes)
            or isinstance(model_or_pipeline, str)
```

Reply:
After testing this, it seems that after this line `from datasets import Dataset, load_dataset`, `"tensorflow" in sys.modules` becomes `True`. And so it doesn't help with the problem.
Is it just my python?

Reply:
Yea it's an issue with `datasets`, let me fix it

Reply:
Actually it comes from `huggingface_hub` - let me open a PR

Reply:
It's already fixed on the `main` branch of `huggingface-hub`. I also added a test here for the future: https://github.com/huggingface/huggingface_hub/pull/904

Reply:
Just tested that it works for `from datasets import Dataset, load_dataset` 👌
But I think `from transformers import Pipeline` in evaluator.py still puts tensorflow in sys.modules

Reply:
This suggestion might avoid the load yes. The current test as it is useless as no model is ever just a `PreTrainedModel` or a `TFPreTrainedModel`. They are always subclasses of those.

Reply:
> The current test as it is useless as no model is ever just a PreTrainedModel or a TFPreTrainedModel. They are always subclasses of those.

indeed, using `isinstance` is required here and it's fine to import PreTrainedModel (and also TFPreTrainedModel if tensorflow has been imported by the user) - see my suggestion above

Reply:
I said: "This suggestion might avoid the load yes." ;-)
The rest of the comment is on the existing code.

Reply:
This is actually not enough to not import TF, since importing `pipeline` (from `transformers.pipelines`) does import TF. Is this expected ?

I wouldn't spend too much time in evaluate trying to not load TF at this point. Users can still provide `USE_TF=0` to not import TF in transformers

Reply:
Yes `pipeline` imports TensorFlow if available to load the default model associated with each pipeline. I would recommend not importing `pipeline` before the time it's necessary.